cobc/tree.c

```diff
@@ -4721,6 +4721,15 @@ validate_indexed_key_field (struct cb_file *f, struct cb_field *records,
 	}
 	k = CB_FIELD_PTR (key_ref);
 
+	/* check collating sequence is not ignored */
+	if (f->collating_sequence != NULL &&
+	    cbak == NULL &&	/* only if no alternate key is given (bit of hack for now) */
```

Comment:
Why not checking the alternative keys in a loop as well? Aren't the collations used for those?
What about split keys?

Reply:
Finding a way to avoid emitting "too much" warnings for the file-level collating sequence seems a little tedious here.
Split keys appear to always be handled as alphanumerics (as per the code).